### PR TITLE
ocamlPackages.phylogenetics: unstable-2020-11-23 -> 0.0.0

### PR DESCRIPTION
**Pull request by bcdarwin**
###### Motivation for this change

Update package and dependencies and enable tests.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [NA] (Package updates) Added a release notes entry if the change is major or breaking
  - [NA] (Module updates) Added a release notes entry if the change is significant
  - [NA] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/phylogenetics/default.nix [hunk 1]
```diff
@@ -1,30 +1,38 @@
...
     owner  = "biocaml";
     repo   = pname;
-    rev    = "e6e10efa0a3a8fd61bf4ab69e91a09549cc1ded6";
-    sha256 = "0pmypzp0rvlpzm8zpbcfkphwnhrpyfwfv44kshvx2f8nslmksh8c";
+    rev    = "v${version}";
+    sha256 = "sha256:0h7ik1zcl10259zn3bh4gj7ax0xdxc4jm6ph9y4i3q9z09g1m2g1";
   };
 
-  minimumOCamlVersion = "4.08";  # e.g., uses Float.min
+  minimumOCamlVersion = "4.08";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/phylogenetics/default.nix:17]:
```suggestion
  minimalOCamlVersion = "4.08";
```

File: pkgs/development/ocaml-modules/phylogenetics/default.nix [hunk 2]
```diff
@@ -1,30 +1,38 @@
...
   useDune2 = true;
 
   src = fetchFromGitHub {
     owner  = "biocaml";
     repo   = pname;
-    rev    = "e6e10efa0a3a8fd61bf4ab69e91a09549cc1ded6";
-    sha256 = "0pmypzp0rvlpzm8zpbcfkphwnhrpyfwfv44kshvx2f8nslmksh8c";
+    rev    = "v${version}";
+    sha256 = "sha256:0h7ik1zcl10259zn3bh4gj7ax0xdxc4jm6ph9y4i3q9z09g1m2g1";
   };
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/phylogenetics/default.nix:15]: Please use the released source code: https://github.com/biocaml/phylogenetics/releases/download/v0.0.0/phylogenetics-0.0.0.tbz
In particular, it includes correct version information.

File: pkgs/development/ocaml-modules/phylogenetics/default.nix [hunk 3]
```diff
@@ -1,30 +1,38 @@
...
 
-  minimumOCamlVersion = "4.08";  # e.g., uses Float.min
+  minimumOCamlVersion = "4.08";
 
-  checkInputs = [ alcotest ];
-  buildInputs = [ menhir ];
-  propagatedBuildInputs = [ angstrom-unix biocaml gnuplot gsl lacaml owl ppx_deriving printbox ];
+  checkInputs = [ alcotest bppsuite ];
+  buildInputs = [ menhir menhirLib ];
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/phylogenetics/default.nix:20]: menhirLib should be propagated.

File: pkgs/development/ocaml-modules/phylogenetics/default.nix [hunk 4]
```diff
@@ -1,30 +1,38 @@
...
+    lacaml
+    ppx_deriving
+    printbox
+  ];
 
-  doCheck = false;  # many tests require bppsuite
+  doCheck = true;
 
   meta = with lib; {
     inherit (src.meta) homepage;
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/phylogenetics/default.nix:34]: Don’t forget to fix the `meta.homepage` when switching to `fetchurl`.

**Comment by bcdarwin** [comment]
Everything resolved, thanks.

**Pull request merged**